app/src/sharedTest/java/org/oppia/android/app/topic/practice/TopicPracticeFragmentTest.kt

```diff
@@ -383,6 +389,91 @@ class TopicPracticeFragmentTest {
+        val topicPracticeFragment = topicFragment.childFragmentManager
+          .findFragmentByTag("f${viewPager.currentItem}") as TopicPracticeFragment
+
+        val args = topicPracticeFragment.arguments?.getProto(
+          TopicPracticeFragment.TOPIC_PRACTICE_FRAGMENT_ARGUMENTS_KEY,
+          TopicPracticeFragmentArguments.getDefaultInstance()
+        )
+        val receivedInternalProfileId = topicPracticeFragment
+          .arguments?.extractCurrentUserProfileId()?.internalId ?: -1
+        val receivedTopicId = checkNotNull(args?.topicId) {
+          "Expected topic ID to be included in arguments for TopicPracticeFragment."
+        }
+
+        assertThat(receivedInternalProfileId).isEqualTo(internalProfileId)
+        assertThat(receivedTopicId).isEqualTo(FRACTIONS_TOPIC_ID)
+      }
+    }
+  }
+
+  @Test
+  fun testTopicPracticeFragment_saveInstanceState_verifyCorrectStateRestored() {
+    launchTopicActivityIntent(
+      internalProfileId = internalProfileId,
+      classroomId = TEST_CLASSROOM_ID_1,
+      topicId = FRACTIONS_TOPIC_ID
+    ).use { scenario ->
+      clickPracticeTab()
+      testCoroutineDispatchers.runCurrent()
+
+      scenario.onActivity { activity ->
+        val topicFragment = activity.supportFragmentManager
+          .findFragmentById(R.id.topic_fragment_placeholder) as TopicFragment
+        val viewPager = topicFragment.requireView()
+          .findViewById<ViewPager2>(R.id.topic_tabs_viewpager)
+        val topicPracticeFragment = topicFragment.childFragmentManager
+          .findFragmentByTag("f${viewPager.currentItem}") as TopicPracticeFragment
+
+        topicPracticeFragment.topicPracticeFragmentPresenter
+          .selectedSubtopicIdList = arrayListOf(1, 2, 3)
+        topicPracticeFragment.topicPracticeFragmentPresenter.skillIdHashMap = hashMapOf(
+          1 to mutableListOf("skill_1"),
+          2 to mutableListOf("skill_2", "skill_3")
+        )
+      }
+
+      scenario.recreate()
+
+      scenario.onActivity { activity ->
+        val topicFragment = activity.supportFragmentManager
+          .findFragmentById(R.id.topic_fragment_placeholder) as TopicFragment
+        val viewPager = topicFragment.requireView()
+          .findViewById<ViewPager2>(R.id.topic_tabs_viewpager)
+        val topicPracticeFragment = topicFragment.childFragmentManager
+          .findFragmentByTag("f${viewPager.currentItem}") as TopicPracticeFragment
+
+        val restoredSelectedSubtopicList = topicPracticeFragment.topicPracticeFragmentPresenter
+          .selectedSubtopicIdList
+        val restoredSkillIdHashMap = topicPracticeFragment.topicPracticeFragmentPresenter
+          .skillIdHashMap
+
+        assertThat(restoredSelectedSubtopicList).containsExactly(1, 2, 3)
+        assertThat(restoredSkillIdHashMap)
+          .containsExactly(
+            1, mutableListOf("skill_1"),
+            2, mutableListOf("skill_2", "skill_3")
+          )
```

Comment:
Here & above: similar to my earlier comment, can we test this by analyzing the UI rather than inspecting fragment state? The latter is rather fragile if the implementation state changes, and technically isn't representative of the thing actually being represented by the values.